Code/Framework/AzToolsFramework/AzToolsFramework/Application/ToolsApplication.cpp

```diff
@@ -310,6 +312,7 @@ namespace AzToolsFramework
                 azrtti_typeid<AzToolsFramework::GlobalPaintBrushSettingsSystemComponent>(),
                 azrtti_typeid<AzToolsFramework::MetadataManager>(),
                 azrtti_typeid<AzToolsFramework::UuidUtilComponent>(),
+                azrtti_typeid<AzFramework::NativeUISystemComponent>(),
```

Comment:
This line can be removed as well, as the Module would be responsible for adding the NativeUiSystemComponent has required and not the application.